### PR TITLE
stm32: ns: Clean up and fixes

### PR DESCRIPTION
**Pull request by erwango**

### CODE REVIEW

**Review by nordicjm — DISMISSED** [review]

**Review by tejlmand — DISMISSED** [review]

**Review by gautierg-st — DISMISSED** [review]

File: soc/st/stm32/Kconfig [hunk 1]
```diff
@@ -39,6 +39,12 @@ config STM32_ENABLE_DEBUG_SLEEP_STOP
 	  effectivly destroys the use-case of `west attach`. Also
 	  SEGGER RTT and similar technologies need this.
 
+config BUILD_WITH_TFM
+	default y if TRUSTED_EXECUTION_NONSECURE
+	help
+	  By default, if we build for a Non-Secure version of the board,
+	  enable building with TF-M as the Secure Execution Environment.
```
Comment (nordicjm) [inline soc/st/stm32/Kconfig:46]: This would go in `Kconfig.defconfig`

Comment (nordicjm) [inline soc/st/stm32/Kconfig:46]: And no help text

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: boards/st/nucleo_l552ze_q/nucleo_l552ze_q_stm32l552xx_ns.dts [hunk 2]
```diff
@@ -19,10 +19,55 @@
...
+		/* Secure image primary slot */
+		slot1_partition: partition@4a000 {
+			label = "image-1";
+			reg = <0x0004a000 DT_SIZE_K(180)>;
+		};
+		/* Non-secure image primary slot */
+		slot1_ns_partition: partition@77000 {
+			label = "image-1-nonsecure";
+			reg = <0x00077000 DT_SIZE_K(36)>;
+		};
```
Comment (nordicjm) [inline boards/st/nucleo_l552ze_q/nucleo_l552ze_q_stm32l552xx_ns.dts:69]: secondary

Comment (erwango) [reply boards/st/nucleo_l552ze_q/nucleo_l552ze_q_stm32l552xx_ns.dts:69]: Good catch, thanks

**Review by nordicjm — APPROVED** [review]

**Review by gautierg-st — APPROVED** [review]

**Pull request merged**